app/src/main/java/com/boosters/promise/ui/promise/PromiseSettingViewModel.kt

```diff
@@ -36,16 +61,25 @@ class PromiseSettingViewModel @Inject constructor(
     }
 
     fun onClickCompletionButton() {
-        viewModelScope.launch {
-            // promise 객체 넣어주면 firestore에 저장 및 수정됨.
-            // promiseRepository.addPromise(promise)
+        val promise = _promiseUiState.value
+        if (promise.title.isEmpty() || promise.time.isEmpty() || promise.destinationName.isEmpty() || promise.date.isEmpty()) {
+            _promiseSettingUiState.update { PromiseSettingUiState.Fail(R.string.promiseSetting_empty) }
+            return
+        }
+        val promiseTime = LocalDateTime.parse("${promise.date} ${promise.time}", dateFormatter)
+        if (promiseTime.isBefore(LocalDateTime.now())) {
+            _promiseSettingUiState.update { PromiseSettingUiState.Fail(R.string.promiseSetting_beforeTime) }
+            return
         }
-    }
-
-    fun onClickDeleteButton() {
         viewModelScope.launch {
-            // promise 객체 넣어주면 firestore에 삭제됨.
-            //promiseRepository.removePromise(promise)
+            val members = promise.members.toMutableList()
+            members.add(myInfo.toUserUiState())
+            promiseRepository.addPromise(promise.copy(members = members).toPromise()).collect {
```

Comment:
```suggestion
            val members = promise.members.plus(myInfo.toUserUiState())
            promiseRepository.addPromise(promise.copy(members = members).toPromise()).collect {
```

MutableList로 바꾸고 add하는 대신, plus를 사용하는 방법도 추천드려요.

Reply:
네 감사합니다!